src/clegoues/genprog4java/Search/Search.java

```diff
@@ -263,41 +263,51 @@ public void mutate(Representation<G> variant) throws GiveUpException {
 
```

Comment:
opefully I'm not too offbase with this mutate function.... it was a very confusing time but after reading the code and messing with the code and then rereading the code, I think the old version of the code was functional in a convoluted way (I was not convinced of that statement when I started changing this mutate function).

I kinda just took a cue from what used to be line 284 (not sure if it was a todo, but it seemed like a todo).

I think the main functional difference now is that a variant will always have `promut` number of mutations, whereas before, some of the locations found would not work, causing mutations < `promut`.